flowmachine/flowmachine/core/query.py

```diff
@@ -190,7 +198,7 @@ def get_query(self):
         try:
             table_name = self.fully_qualified_table_name
             schema, name = table_name.split(".")
-            with rlock(self.redis, self.md5):
+            if QueryStateMachine(self.redis, self.md5).wait_until_complete():
```

Comment:
This is confusing to the reader because `wait_until_complete()` sounds like an instruction which I wouldn't expect to return anything. Can we split this into two separate instructions?
```suggestion
            query_state_machine = QueryStateMachine(self.redis, self.md5)
            query_state_machine.wait_until_complete()
            if query_state_machine.is_completed:
```